### PR TITLE
Issue #1121: Filter cross-system direct trips in trip search

### PR DESCRIPTION
**Pull request by trackrat-dev**
## Summary

Closes #1121

- **Added `_filter_cross_system_direct_trips()`** in `trip_search.py` that filters direct trip results to only keep trips whose `data_source` natively serves both the from and to station codes. This prevents station equivalence expansion (e.g., `{NY, S128, SA28}`) from causing cross-system matches to appear as "direct" trips.
- **Moved `from_systems`/`to_systems` computation** earlier in `search_trips()` so it's computed once and reused for both direct-trip filtering and the transfer search.
- When all direct trips are filtered (e.g., `from=TR to=S128` where no system serves both), the function correctly falls through to the transfer search.

## Files modified

| File | Change |
|------|--------|
| `backend_v2/src/trackrat/services/trip_search.py` | Added `_filter_cross_system_direct_trips()` function; integrated it into `search_trips()` between direct trip construction and the early-return check |
| `backend_v2/tests/unit/services/test_trip_search.py` | Added `TestFilterCrossSystemDirectTrips` class with 12 test cases covering: empty input, same-system passthrough, cross-system filtering, disjoint systems, mixed keep/filter, intra-subway, reverse direction, all three cross-system hub groups (Penn/GCT/WTC), and integration tests with real station topology data |

## Design decisions

- **Filter in `search_trips`, not in `expand_station_codes` or `get_departures`**: The equivalence expansion is correct for departure boards — only trip search needs the cross-system filter. This keeps the change minimal and avoids touching shared infrastructure.
- **Uses `get_systems_serving_station()` intersection**: A trip is "direct" only if its `data_source` is in `from_systems ∩ to_systems`. This naturally handles all cross-system hub equivalence groups (Penn Station, Grand Central, WTC/Oculus, Newark Penn/PATH, Chicago Union).
- **`data_sources` user filter is applied separately**: The cross-system filter uses the full system sets (station-level truth), while the user's `data_sources` preference is applied later for the transfer search only. The departure service already respects `data_sources` for its query.

## Test coverage

12 new tests in `TestFilterCrossSystemDirectTrips`:
- `test_empty_list_returns_empty` — edge case
- `test_same_system_keeps_all` — NJT→NJT passthrough
- `test_cross_system_filtered_out` — core bug: TR→S128 with AMTRAK/NJT trips
- `test_disjoint_systems_returns_empty` — AMTRAK→SUBWAY
- `test_shared_system_kept_different_system_filtered` — NY→TR keeps NJT/AMTRAK, filters LIRR
- `test_subway_to_subway_direct` — intra-system passthrough
- `test_penn_station_subway_to_trenton_filtered` — reverse direction S128→TR
- `test_grand_central_mnr_to_subway_filtered` — GCT→S631
- `test_wtc_path_to_subway_filtered` — PWC→S138
- `test_uses_real_station_systems` — integration with real topology data for TR/S128
- `test_ny_to_trenton_keeps_njt_amtrak` — sanity check: normal direct route unaffected

All 76 trip search tests pass. Full unit suite: 2469 passed (217 errors are pre-existing DB-connection failures unrelated to this change).

https://claude.ai/code/session_01HSXxXGew7LAcybbJ8vPj2r

---
_Generated by [Claude Code](https://claude.ai/code/session_01HSXxXGew7LAcybbJ8vPj2r)_

### CODE REVIEW

File: backend_v2/src/trackrat/services/trip_search.py [hunk 1]
```diff
@@ -267,6 +288,11 @@ async def search_trips(
         data_sources=data_sources,
     )
 
+    # Systems natively serving each endpoint (used for direct-trip filtering
+    # and transfer search below).
+    from_systems = get_systems_serving_station(from_station)
+    to_systems = get_systems_serving_station(to_station)
```
[the remark this answers is not in the document]
Comment (trackrat-dev) [reply backend_v2/src/trackrat/services/trip_search.py]: Valid concern — fixed in 99e6956.

`from_systems`/`to_systems` are now computed by expanding station codes via equivalence groups first:

```python
from_systems: set[str] = set()
for code in expand_station_codes(from_station):
    from_systems |= get_systems_serving_station(code)
```

This ensures alias-only codes like `TS` resolve to `{"NJT"}` via the canonical `SE` code. Added a test (`test_alias_code_resolves_via_equivalence_expansion`) that verifies `TS` resolves correctly and NJT direct trips are preserved.

---
_Generated by [Claude Code](https://claude.ai/code)_
